diff --git a/emsdk/Makefile b/emsdk/Makefile
index 0c31e980..5e98440a 100644
--- a/emsdk/Makefile
+++ b/emsdk/Makefile
@@ -6,6 +6,7 @@ all: emsdk/.complete
 emsdk/.complete:
 	if [ -d emsdk ]; then rm -rf emsdk; fi
 	git clone https://github.com/juj/emsdk.git
+	(cd emsdk && git checkout 3.1.29)
 	( \
 		cd emsdk && \
 		./emsdk install --build=Release sdk-fastcomp-tag-$(EMSCRIPTEN_VERSION)-64bit binaryen-tag-$(EMSCRIPTEN_VERSION)-64bit && \
